sql/core/src/test/scala/org/apache/spark/sql/MathFunctionsSuite.scala

```diff
@@ -707,6 +707,11 @@ class MathFunctionsSuite extends QueryTest with SharedSparkSession {
       df1.select(try_divide(make_interval(col("year"), col("month")), lit(0))))
   }
 
+  test("try_remainder") {
+    val df = Seq((10, 3), (5, 5), (5, 0)).toDF("birth", "age")
```

Comment:
Let's have some end-to-end test with decimal input as well, either in this test suite or in `sql/core/src/test/resources/sql-tests/inputs/try_arithmetic.sql`

context: there was a bug found recently https://github.com/apache/spark/pull/46286

Reply:
Done